pageserver/src/tenant/remote_timeline_client.rs

```diff
@@ -393,22 +397,27 @@ impl RemoteTimelineClient {
         layer_file_name: &LayerFileName,
         layer_metadata: &LayerFileMetadata,
     ) -> anyhow::Result<u64> {
-        let downloaded_size = download::download_layer_file(
-            self.conf,
-            &self.storage_impl,
-            self.tenant_id,
-            self.timeline_id,
-            layer_file_name,
-            layer_metadata,
-        )
-        .measure_remote_op(
-            self.tenant_id,
-            self.timeline_id,
-            RemoteOpFileKind::Layer,
-            RemoteOpKind::Download,
-            Arc::clone(&self.metrics),
-        )
-        .await?;
+        let downloaded_size = {
+            let _unfinished_gauge_guard = self
+                .metrics
+                .call_begin(&RemoteOpFileKind::Layer, &RemoteOpKind::Download);
+            download::download_layer_file(
+                self.conf,
```

Comment:
can't this be a struct within RemoteClientDownload that is shared with download.rs ?

Reply:
What is `this`? 

Note that this PR didn't change anything about `download::download_layer_file`. I just had to indent it into a new scope so that we drop the (new) `_unfinished_gauge_guard` in a timely manner.

Reply:
What I meant was the download::download_layer_file will receive RemoteClientDownload as a parameter

Reply:
Sure it could - I guess this comes from the angle of 'long parameter lists are hard to read'.
I personally think 6 parameters is fine iff they all have different types.
That is the case here.